File: Server/src/main/java/com/example/server/entity/Event.java
package com.example.server.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

/**
 * Entity representing an event in the ticketing system.
 * This class stores the event details, such as the event name, ticket price, capacity, and sold-out ticket count.
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @NonNull
    @Column(nullable = false)
    private String eventName;

    @Column(nullable = false)
    private float ticketPrice;

    @Column(nullable = false)
    private int maxCapacity;

    @Column(nullable = false)
    private int soldOutTickets;
}

File: client/src/app/model/class/Event.ts
/**
 * Represents an event, including details such as the event name, ticket price,
 * maximum capacity, and the number of tickets that have been sold out.
 */
export class Event{
    /** Unique identifier for the event. */
    id: number;

    /** Name of the event. */
    eventName: string;

    /** Price of a single ticket for the event. */
    ticketPrice: number;

    /** Maximum capacity of tickets available for the event. */
    maxCapacity: number;

    /** Number of tickets that have been sold out for the event. */
    soldOutTickets: number;

    /**
     * Initializes a new Event instance with default values.
     */
    constructor() {
        this.id = 0;
        this.eventName = '';
        this.ticketPrice = 0;
        this.maxCapacity = 0;
        this.soldOutTickets = 0;
    }
}